sycl/source/detail/queue_impl.hpp

```diff
@@ -388,6 +388,10 @@ class queue_impl {
   template <typename Param>
   typename Param::return_type get_backend_info() const;
 
+  /// Begin to execute previously issued commands on this queue
+  /// if they have not been executed yet. Overrides normal batching behaviour.
+  void flush() { getPlugin()->call<PiApiKind::piQueueFlush>(MQueues[0]); }
```

Comment:
Also, I think special handling is needed for the case when graph recording is in progress. (for example, see queue_impl::wait).  It will be nice to have test for this case as well.